test/widgets/autocomplete_test.dart

```diff
@@ -177,6 +177,71 @@ void main() {
 
       debugNetworkImageHttpClientProvider = null;
     });
+
+    testWidgets('user options appear in the correct rendering order and do not scroll down', (tester) async {
+      final List<User> users = [
+        eg.user(userId: 1, fullName: 'Aaditya', avatarUrl: 'user1.png'),
+        eg.user(userId: 2, fullName: 'Alya', avatarUrl: 'user2.png'),
+        eg.user(userId: 3, fullName: 'Aman', avatarUrl: 'user3.png'),
+        eg.user(userId: 4, fullName: 'Anders', avatarUrl: 'user4.png'),
+        eg.user(userId: 5, fullName: 'Anthony', avatarUrl: 'user5.png'),
+        eg.user(userId: 6, fullName: 'Apoorva', avatarUrl: 'user6.png'),
+        eg.user(userId: 7, fullName: 'Asif', avatarUrl: 'user7.png'),
+        eg.user(userId: 8, fullName: 'Asim', avatarUrl: 'user8.png')];
+
+      final composeInputFinder = await setupToComposeInput(tester, users: users);
+      final store = await testBinding.globalStore.perAccount(eg.selfAccount.id);
+
+      // Options are filtered correctly for query
+      // TODO(#226): Remove this extra edit when this bug is fixed.
+      await tester.enterText(composeInputFinder, 'hello @');
+      await tester.enterText(composeInputFinder, 'hello @A');
+      await tester.pump();
+      // Only first seven users render initially, 8th user has to be accessed by scrolling up
+      final List<double> positions = [];
+      for (int i = 0; i < 7; i++) {
+        final user = users[i];
+        checkUserShown(user, store, expected: true);
+        positions.add(tester.getTopLeft(find.text(user.fullName)).dy);
+      }
+      for (int i = 7; i < users.length; i++) {
+        final user = users[i];
+        checkUserShown(user, store, expected: false);
+      }
+
+      final listViewFinder = find.byType(ListView);
+      await tester.drag(listViewFinder, const Offset(0, -50));
+      await tester.pump();
+
+      final firstMentionPositionAfterDragDown = tester.getTopLeft(find.text(users.first.fullName)).dy;
+      check(
+        because:'ListView options should not scroll down because they are already at the bottom',
+        positions[0]
+      ).equals(firstMentionPositionAfterDragDown);
+
+      for (int i = 0; i < 6; i++) {
+         check(
+          because: """${users[i + 1]} should appear above
+          ${users[i]} because of reverse order""",
+          positions[i]).isGreaterThan(positions[i+1]);
+      }
+
+      await tester.drag(listViewFinder, const Offset(0, 200)); // Should be capped at prev position
+      await tester.pump();
+
+      checkUserShown(users.last, store, expected: true);
+      checkUserShown(users.first, store, expected: false);
+
+      // 8th user should be above 7th user
+      check(because: "8th user should be above 7th user",
+      tester.getTopLeft(
+        find.text(users.last.fullName)).dy
+      ).isLessThan(
+        tester.getTopLeft(
+        find.text(users[users.length - 2].fullName)
+      ).dy);
+      debugNetworkImageHttpClientProvider = null;
+    });
```

Comment:
The formatting looks good, but I think this test can be simplified.

The purpose our testing is to make sure we don't break a feature change. In this case, it is the change that the autocomplete options are reversed. So if the test fails when I remove the `reversed: true` line, which it does, the test is effective.

However, it would also be nice if we can simplify the test. Currently, it does 3 checks:
1. verify that initially, the first 7 options are shown and we cannot scroll down; 2. check that the options appear before one another, matching the expected order; 3. check that scrolling up, the last option is shown and the first one is hidden.

Among them, I think only 1 and 3 are necessary for the sake of testing `reversed: true`. They happen to check scrolling up and down, which are quite relevant to this feature. 2 would enforce the test at the cost of making it a bit more complicated, but the test is already sufficient without it.

To make the test more readable, we should try to minimize lines of code that set up the test, and highlight those that are more interesting, like the ones checking if an option is visible or not. Here's an updated version of this test:

```dart
    testWidgets('options are shown in reversed order', (tester) async {
      final users = List.generate(8, (i) => eg.user(fullName: 'A$i', avatarUrl: 'user$i.png'));
      final composeInputFinder = await setupToComposeInput(tester, users: users);
      final store = await testBinding.globalStore.perAccount(eg.selfAccount.id);

      // TODO(#226): Remove this extra edit when this bug is fixed.
      await tester.enterText(composeInputFinder, 'hello @');
      await tester.enterText(composeInputFinder, 'hello @A');
      await tester.pump();

      final initialPosition = tester.getTopLeft(find.text(users.first.fullName)).dy;
      // Initially, all but the last autocomplete options are visible.
      checkUserShown(users.last, store, expected: false);
      users.take(7).forEach((user) => checkUserShown(user, store, expected: true));

      // Can't scroll down because the options grow from the bottom.
      await tester.drag(find.byType(ListView), const Offset(0, -50));
      await tester.pump();
      check(tester.getTopLeft(find.text(users.first.fullName)).dy)
        .equals(initialPosition);

      // The last autocomplete option becomes visible after scrolling up.
      await tester.drag(find.byType(ListView), const Offset(0, 200));
      await tester.pump();
      users.skip(1).forEach((user) => checkUserShown(user, store, expected: true));
      checkUserShown(users.first, store, expected: false);

      debugNetworkImageHttpClientProvider = null;
    })
```